x-pack/plugins/security_solution/server/lib/siem_migrations/rules/api/get_prebuilt_rules.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { IKibanaResponse, Logger } from '@kbn/core/server';
+import { buildRouteValidationWithZod } from '@kbn/zod-helpers';
+import type { GetRuleMigrationPrebuiltRulesResponse } from '../../../../../common/siem_migrations/model/api/rules/rule_migration.gen';
+import { GetRuleMigrationPrebuiltRulesRequestParams } from '../../../../../common/siem_migrations/model/api/rules/rule_migration.gen';
+import { SIEM_RULE_MIGRATIONS_PREBUILT_RULES_PATH } from '../../../../../common/siem_migrations/constants';
+import type { SecuritySolutionPluginRouter } from '../../../../types';
+import { withLicense } from './util/with_license';
+import { getPrebuiltRules } from './util/prebuilt_rules';
+import { MAX_PREBUILT_RULES_TO_FETCH } from './constants';
+
+export const registerSiemRuleMigrationsPrebuiltRulesRoute = (
+  router: SecuritySolutionPluginRouter,
+  logger: Logger
+) => {
+  router.versioned
+    .get({
+      path: SIEM_RULE_MIGRATIONS_PREBUILT_RULES_PATH,
+      access: 'internal',
+      security: { authz: { requiredPrivileges: ['securitySolution'] } },
+    })
+    .addVersion(
+      {
+        version: '1',
+        validate: {
+          request: {
+            params: buildRouteValidationWithZod(GetRuleMigrationPrebuiltRulesRequestParams),
+          },
+        },
+      },
+      withLicense(
+        async (
+          context,
+          req,
+          res
+        ): Promise<IKibanaResponse<GetRuleMigrationPrebuiltRulesResponse>> => {
+          const { migration_id: migrationId } = req.params;
+          try {
+            const ctx = await context.resolve(['core', 'alerting', 'securitySolution']);
+            const ruleMigrationsClient = ctx.securitySolution.getSiemRuleMigrationsClient();
+            const savedObjectsClient = ctx.core.savedObjects.client;
+            const rulesClient = await ctx.alerting.getRulesClient();
+
+            const result = await ruleMigrationsClient.data.rules.get(
+              {
+                migrationId,
+                prebuiltRulesOnly: true,
+              },
+              0,
+              MAX_PREBUILT_RULES_TO_FETCH
```

Comment:
Not sure about that, what happens if the migration has more rules than `MAX_PREBUILT_RULES_TO_FETCH`?

Reply:
We can always put 10k as a maximum search number for this constant? Or do we want to be able to work with any number of docs here?

Reply:
yes, right? we need to take all the rules that match a prebuilt rule, not only a subset. If we limit the ES `get` results we need to iterate that to retrieve all of them, but I think we can just remove the limit here.

Reply:
If we remove it here, it will fetch only 10 docs.

[Size docs](https://www.elastic.co/guide/en/elasticsearch/reference/8.15/search-search.html):
```
(Optional, integer) Defines the number of hits to return. Defaults to 10.

By default, you cannot page through more than 10,000 hits using the from and size parameters. To page through more hits, use the [search_after](https://www.elastic.co/guide/en/elasticsearch/reference/8.15/paginate-search-results.html#search-after) parameter.
```

Reply:
Oh, yeah ES has a default size of 10 🤦 Sorry I forgot.
Okay, let's create a `data.rules.getAll` that iterates until everything is retrieved. I also need that for the resources. If we always scope the search by migration, so having rule documents in the order of 10^3 should not be a problem. 

I'll implement it on my side, if you want you can implement it as well :) or just use a limit of 10.000 temporarily.

Reply:
I will go with the 10k limit for now and we can extend that functionality in a separate PR. I can take care of it